core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala

```diff
@@ -88,6 +88,18 @@ private[spark] class ShuffleMapStage(
    */
   def isAvailable: Boolean = numAvailableOutputs == numPartitions
 
+  /**
+   * Retruns true if push based shuffle is disabled for this stage, or if the shuffle merge for
+   * this stage is finalized, i.e. the shuffle merge results for all partitions are available.
+   */
+  def isMergeFinalized: Boolean = {
```

Comment:
This API is not used in this patch. Should we include it?
Merge finalized is not used at the beginning of a shuffle map stage.